indico/web/client/js/react/components/WTFDateTimeField.jsx

```diff
@@ -0,0 +1,237 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import 'react-dates/initialize';
+import TimePicker from 'rc-time-picker';
+import PropTypes from 'prop-types';
+import React, {useEffect, useMemo, useRef, useState, useCallback} from 'react';
+import moment from 'moment';
+import {SingleDatePicker} from 'indico/react/components';
+import {Translate} from 'indico/react/i18n';
+
+import 'rc-time-picker/assets/index.css';
+import 'react-dates/lib/css/_datepicker.css';
+
+function trigger(id) {
+  document.getElementById(id).dispatchEvent(new Event('change', {bubbles: true}));
+}
+
+export default function WTFDateTimeField({
+  timeId,
+  dateId,
+  timezoneFieldId,
+  timezone,
+  uses24HourFormat,
+  required,
+  disabled,
+  allowClear,
+  earliest,
+  latest,
+  defaultTime,
+  linkedField,
+}) {
+  const dateField = useMemo(() => document.getElementById(dateId), [dateId]);
+  const timeField = useMemo(() => document.getElementById(timeId), [timeId]);
+  const timezoneField = useMemo(() => document.getElementById(timezoneFieldId), [timezoneFieldId]);
+  const linkedFieldDateElem = useMemo(
+    () => linkedField && document.getElementById(`${linkedField.id}-datestorage`),
+    [linkedField]
+  );
+  const [time, setTime] = useState(timeField.value ? moment(timeField.value, 'HH:mm') : null);
+  const [date, setDate] = useState(dateField.value ? moment(dateField.value, 'DD/MM/YYYY') : null);
+  const [currentTimezone, setTimezone] = useState(timezone);
+  const earliestMoment = earliest ? moment(earliest) : null;
+  const latestMoment = latest ? moment(latest) : null;
+  const format = uses24HourFormat ? 'HH:mm' : 'hh:mm a';
+  const timePickerRef = useRef();
+  const clearRef = useRef();
+
+  const updateTime = useCallback(
+    value => {
+      timeField.value = value ? value.format('HH:mm') : '';
+      setTime(value);
+      trigger(timeId, 'change');
+    },
+    [timeField, timeId]
+  );
+
+  const updateDate = useCallback(
+    value => {
+      if (value && timeField.value === '') {
+        updateTime(moment(defaultTime, 'HH:mm'));
+      }
+      dateField.value = value ? value.format('DD/MM/YYYY') : '';
+      setDate(value);
+      trigger(dateId, 'change');
+      timePickerRef.current.picker.required = value;
+    },
+    [dateField, timeField, dateId, defaultTime, updateTime]
+  );
+
+  useEffect(() => {
+    if (!timezoneField) {
+      return;
+    }
+    function handleTimezone() {
+      setTimezone(timezoneField.value);
+    }
+    timezoneField.addEventListener('change', handleTimezone);
+
+    return () => {
+      timezoneField.removeEventListener('change', handleTimezone);
+    };
+  }, [timezoneField]);
+
+  useEffect(() => {
+    timePickerRef.current.picker.required = required;
```

Comment:
it doesn't have one, i was looking into it before suggesting her this ugly workaround as well :D

Reply:
I made a quick-test with required=true and got the HTML default pop-up. Not sure if that is sufficient though.

Reply:
did you comment out both occurrences of `timePickerRef.current.picker.required`? because I didn't find anything with required in https://github.com/react-component/time-picker/blob/master/src/TimePicker.jsx

Reply:
My bad, not all the occurrences were commented, I assumed the props were being passed through 😄